src/socket.io/posts/tools.js

```diff
@@ -172,6 +172,14 @@ module.exports = function (SocketPosts) {
 			throw new Error('[[error:no-privileges]]');
 		}
 
-		await posts.changeOwner(data.pids, data.toUid);
+		var postData = await posts.changeOwner(data.pids, data.toUid);
+		postData = postData.map(({ pid, tid, uid, cid, timestamp }) => ({ pid, tid, uid, cid, timestamp }));
+		await events.log({
+			type: 'post-change-owner',
+			uid: socket.uid,
+			ip: socket.ip,
+			newOwner: data.toUid,
+			originalPosts: postData,
```

Comment:
Don't think this will work in redis since it doesn't support nested objects/arrays.

Reply:
Hmm... Didn't think of that. Would making it into an object with pid as the key work?

Reply:
Nope, the object you pass into events.log is gets sent in https://github.com/NodeBB/NodeBB/blob/master/src/events.js#L78 so you can't have other objects/arrays in it. You can turn the postData to json string if you really want to save it. Or you would have to store an event for each post.

Reply:
Decided to create an event for each post. I think knowing who was the original author is useful here and it's unfortunate redis doesn't allow you to store this easily.